thingif/src/test/java/com/kii/thingif/thingifapi/BuilderTest.java

```diff
@@ -0,0 +1,149 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RunWith(RobolectricTestRunner.class)
+public class BuilderTest extends ThingIFAPITestBase{
+    private Context context;
+
+    @Before
+    public void before() throws Exception{
+        this.context = RuntimeEnvironment.application.getApplicationContext();
+    }
+    @Test
+    public void basicTest() throws Exception {
+        Map<String, Class<? extends Action>> actionTypes = new HashMap<>();
+        actionTypes.put("airConditionerAlias", AirConditionerActions.class);
+        actionTypes.put("humidityAlias", HumidityActions.class);
+
+        ThingIFAPI.Builder builder = ThingIFAPI.Builder.newBuilder(
+                this.context,
+                new KiiApp("appid", "appkey", Site.JP),
+                new Owner(new TypedID(TypedID.Types.USER, "user1234"), "token"),
+                getDefaultActionTypes(),
+                getDefaultStateTypes());
+        ThingIFAPI api = builder.build();
+
+        Assert.assertEquals("appid", api.getAppID());
+        Assert.assertEquals("appkey", api.getAppKey());
+        Assert.assertEquals(Site.JP.getBaseUrl(), api.getBaseUrl());
+        Assert.assertEquals(new TypedID(TypedID.Types.USER, "user1234"), api.getOwner().getTypedID());
+        Assert.assertEquals("token", api.getOwner().getAccessToken());
+        Assert.assertEquals(api.getActionTypes(), getDefaultActionTypes());
+        Assert.assertEquals(api.getStateTypes(), getDefaultStateTypes());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void newBuilderWithNullContextTest() throws Exception {
+        ThingIFAPI.Builder.newBuilder(
+                null, 
+                new KiiApp("appid", "appkey", Site.JP), 
+                new Owner(new TypedID(TypedID.Types.USER, "user1234"), "token"));
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void newBuilderWithNullAppIDTest() throws Exception {
+        ThingIFAPI.Builder.newBuilder(
+                this.context,
+                new KiiApp(null, "appkey", Site.JP),
+                new Owner(new TypedID(TypedID.Types.USER, "user1234"), "token"));
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void newBuilderWithEmptyAppIDTest() throws Exception {
+        ThingIFAPI.Builder.newBuilder(
+                this.context,
+                new KiiApp("", "appkey", Site.JP),
+                new Owner(new TypedID(TypedID.Types.USER, "user1234"), "token"));
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void newBuilderWithNullAppKeyTest() throws Exception {
+        ThingIFAPI.Builder.newBuilder(
+                this.context,
+                new KiiApp("appid", null, Site.JP),
+                new Owner(new TypedID(TypedID.Types.USER, "user1234"), "token"));
```

Comment:
This is Test for `KiiApp` constructor. `ThingIFAPI.Builder` is needless.

What do you think?